pkg/object/httpserver/mux.go

```diff
@@ -349,6 +349,10 @@ func (m *mux) reloadRules(superSpec *supervisor.Spec, muxMapper protocol.MuxMapp
 	} else if oldRules.tracer != nil {
 		tracer = oldRules.tracer
 	}
+	super := superSpec.Super()
+	if super != nil {
+		super.SetTracing(tracer) // share tracing object with proxy filters
+	}
```

Comment:
please check if there's a race condition with the call to `super.SetTracing`.

Reply:
Good catch! Changing supervisor.tracing to `atomic.Value` should fix race conditions